source/module_io/get_wf_lcao.cpp

```diff
@@ -475,27 +415,219 @@ void IState_Envelope::begin(const psi::Psi<std::complex<double>>* psi,
+            {
+                if (ib < fermi_band + bands_above)
+                {
+                    bands_picked_[ib] = 1;
+                }
+            }
+        }
+    }
+    else if (mode == 2)
+    {
+        // Check if length of out_wfc_kb is valid
+        if (static_cast<int>(out_wfc_kb.size()) > nbands)
+        {
+            ModuleBase::WARNING_QUIT("IState_Envelope::select_bands",
+                                     "The number of bands specified by `out_wfc_norm` or `out_wfc_re_im` in the INPUT "
+                                     "file exceeds `nbands`!");
+        }
+        // Check if all elements in out_wfc_kb are 0 or 1
+        for (int value: out_wfc_kb)
+        {
+            if (value != 0 && value != 1)
+            {
+                ModuleBase::WARNING_QUIT(
+                    "IState_Envelope::select_bands",
+                    "The elements of `out_wfc_norm` or `out_wfc_re_im` must be either 0 or 1. Invalid values found!");
+            }
+        }
+        // Fill bands_picked_ with values from out_wfc_kb
+        // Remaining bands are already set to 0
+        const int length = std::min(static_cast<int>(out_wfc_kb.size()), nbands);
+        for (int i = 0; i < length; ++i)
+        {
+            // out_wfc_kb rely on function parse_expression
+            bands_picked_[i] = out_wfc_kb[i];
+        }
+
+        // Check if there are selected bands below the Fermi surface
+        bool has_below = false;
+        for (int i = 0; i + 1 <= fermi_band; ++i)
+        {
+            if (bands_picked_[i] == 1)
+            {
+                has_below = true;
+                break;
+            }
+        }
+        if (has_below)
+        {
+            std::cout << " Plot wave functions below the Fermi surface: band ";
+            for (int i = 0; i + 1 <= fermi_band; ++i)
+            {
+                if (bands_picked_[i] == 1)
+                {
+                    std::cout << i + 1 << " ";
+                }
+            }
+            std::cout << std::endl;
+        }
+
+        // Check if there are selected bands above the Fermi surface
+        bool has_above = false;
+        for (int i = fermi_band; i < nbands; ++i)
+        {
+            if (bands_picked_[i] == 1)
+            {
+                has_above = true;
+                break;
+            }
+        }
+        if (has_above)
+        {
+            std::cout << " Plot wave functions above the Fermi surface: band ";
+            for (int i = fermi_band; i < nbands; ++i)
+            {
+                if (bands_picked_[i] == 1)
+                {
+                    std::cout << i + 1 << " ";
+                }
+            }
+            std::cout << std::endl;
+        }
+    }
+    else
+    {
+        ModuleBase::WARNING_QUIT("IState_Envelope::select_bands", "Invalid mode! Please check the code.");
+    }
+}
+
 // for each band
 void IState_Envelope::set_pw_wfc(const ModulePW::PW_Basis_K* wfcpw,
```

Comment:
I think it is quite confusing about the nomenclature of variables here, but I know it is not because of your changes.
There is obvious inconsistency between names of ptr to PW_Basis_K instance and the ref of instance of psi. The latter is wavefunction in g space, so it is okay to name it as "wfc_g" but it is nonsence to name instance of PW_Basis_K as wfcpw, because it is not the wfc itself at all...